exonum/src/blockchain/config.rs

```diff
@@ -23,50 +30,81 @@ use storage::StorageValue;
 use helpers::{Height, Milliseconds};
 
-/// Public keys of a validator.
+/// Public keys of a validator. Each validator has two public keys: the
+/// `consensus_key` is used for internal operations in the consensus process,
+/// while the `service_key` is used in services.
 #[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct ValidatorKeys {
     /// Consensus key is used for messages related to the consensus algorithm.
-    #[doc(hidden)]
     pub consensus_key: PublicKey,
-    /// Service key is used for services.
+    /// Service key is used for services, for example, the configuration
+    /// updater service, the anchoring service, etc.
     pub service_key: PublicKey,
 }
 
-/// Exonum blockchain global configuration.
-/// This configuration must be same for any exonum node in the certain network on given height.
+/// Exonum blockchain global configuration. Services
+/// and their parameters are also included into this configuration.
+///
+/// This configuration must be the same for any Exonum node in a certain
+/// network on the given height.
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct StoredConfiguration {
-    /// Link to the previous configuration.
-    /// For configuration in the genesis block `hash` is just an array of zeros.
+    /// Hash of the previous configuration, which can be used to find that
+    /// configuration. For the configuration in the genesis block,
+    /// `hash` is just an array of zeros.
     pub previous_cfg_hash: Hash,
-    /// The height, starting from which this configuration becomes actual.
+    /// The height, starting from which this configuration becomes actual. Note
+    /// that this height should be big enough for the nodes to accept the new
+    /// configuration before this height is reached. Otherwise, the new
+    /// configuration will not take effect at all; the old configuration will
+    /// remain actual.
     pub actual_from: Height,
-    /// List of validators' consensus and service public keys.
+    /// List of validator consensus and service public keys.
     pub validator_keys: Vec<ValidatorKeys>,
     /// Consensus algorithm parameters.
     pub consensus: ConsensusConfig,
-    /// Number of votes required to commit new configuration.
-    /// Should be greater than 2/3 and less or equal to the validators count.
+    /// Number of votes required to commit the new configuration.
+    /// This value should be greater than 2/3 and less or equal to the
+    /// validators count.
     pub majority_count: Option<u16>,
     /// Services specific variables.
-    /// Keys are `service_name` from `Service` trait and values are the serialized json.
+    /// Keys are `service_name` from the `Service` trait and values are the serialized json.
     #[serde(default)]
     pub services: BTreeMap<String, serde_json::Value>,
 }
 
 /// Consensus algorithm parameters.
+///
+/// This configuration is initially created with default recommended values.
```

Comment:
This sentence doesn't make sense to me: it's possible to create a configuration with any values of fields.